lib/src/package.dart

```diff
@@ -496,3 +507,21 @@ class PackageDep extends PackageName {
       samePackage(other) &&
       other.constraint == constraint;
 }
+
+/// The type of dependency from one package to another.
+class DependencyType {
```

Comment:
Dart's enums are so broken for so many use-cases that I've never bothered to learn how they work. Can you provide different stringifications for each value?